### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by Lastin** [buggins/dlangui#231]
`0x0000000000762449 in _D7dlangui4core5types46__T3RefTC7dlangui8graphics9resources8DrawableZ3Ref5clearMFZv (this=...) at src/dlangui/core/types.d:429`

I've tried tracking the cause of this error for long time, but it appears so randomly it's hard to tell.
My only guess is that it might be caused by calling `window.removePopup(_currentPopup)` while drawing, but not sure.

Call stack:
[output1.txt](https://github.com/buggins/dlangui/files/213123/output1.txt)

Error occurs in my project using this library:
https://github.com/Lastin/AlienCivilisations

**Comment by Lastin** [comment]
I have been able to recreate it almost every time by:
1. Starting the program
2. Selecting _Tutorial_ from menu
3. Clicking _Exit Tutorial_
4. Selecting _New Game_ from menu
Wait few seconds and it should throw an error.

I don't know what the library does, but simply describing what my code does:
1. Object of ViewHandler is created
2. ViewHandler calls `window.mainWidget = new Menu()`
3. View Hanlder calls `window.mainWidget = new Tutorial()` (Tutorial extends Play)
4. Tutorial calls `window.showPopup(some widget)`
5. Tutorial calls `window.removePopup(widgetFromPoint4)` and `widgetFromPoint4.destroy()`
6. ViewHandler calls `window.mainWidget = new Menu()` and `tutorial.destroy()`
7. ViewHanlder calls `window.mainWidget = new Play()`

**Comment by buggins** [comment]
Possible root cause: when mainWidget is changed it becomes available for GC.
It's not supposed that widgets will be collected by GC because it breaks resource management (RAII).

Please check if recent fix resolves your crash.

**Comment by Lastin** [comment]
I think it fixed it. Thank you!
Now calling the `window.showPopup(new popup)` caused crashing if a widget that calls that function was not set as window's main widget first. But work-around was quite simple. Instead of:
```
_window.mainWidget = new SomeLayout(_window);
SomeLayout {
  this(Window window) {
    window.showPopup( some popup )
  }
}
```
I did 
```
SomeLayout tut = new SomeLayout();
SomeLayout {
  void func() {
    window.showPopup(somePopup)
  }
}
_window.mainWidget = tut;
tut.func();
```

Strangely enough error was different if I was the passing reference to window into the class constructor and using that reference to show a new popup, and different if I used public variable of window after an object of mentioned class has been initialised.

I also get some segmentation fault in ftfonts.d in function `clear() { FT_Done_Face(_face) }`, but I will fill separate issue describing it.
